addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -4015,24 +4015,248 @@ def test_accrual_allocation_date_in_the_future(self):
+
+            # Right after spending all the 10 leaves ('2023-01-02' -> '2023-01-13')
+            # 10 days - 10 days (leave) + 2 days (1 month accrual)
+            self.assert_virtual_leaves_equal(leave_type_day, 2, self.employee_emp, date='2023-02-01')
+
+    def test_get_allocation_future_leaves1(self):
+        self._test_get_allocation_future_leaves(add_future_leave=False)
+
+    def test_get_allocation_future_leaves2(self):
+        self._test_get_allocation_future_leaves(add_future_leave=True)
+
+    def _test_get_allocation_future_leaves_regular(self, add_future_leave, regular_before):
+        """
+        Context:
+            1) Allocation with a 1 level accrual plan which
+                - Carry all days over
+                - Accrues monthly at the end of the period
+                - Has maximum 10 leaves
+            2) Regular allocation
+        Assert the virtual remaining leaves of the employee allocation are not frozen while taking multiple leaves
+        and using `get_allocation_data` with the `target_date` parameter set in the future.
+        :param add_future_leave: adds another leave in the future (which shouldn't be taken into account) or not
+        :param regular_before: set the `date_from` of the regular allocation before the `date_from` of the accrual allocation
+        """
+
+        with freeze_time('2022-01-01'):
+            accrual_plan = self._get_accrual_plan_sample0(accrued_days=2, accrued_gain_time='end', max_leaves=10)
+            leave_type_day = self._get_leave_type_day_sample0()
+            accrual_allocation = self._form_get_allocation_sample(leave_type_day, '2019-01-01', accrual_plan)
+            accrual_allocation._update_accrual()
+            regular_alloc_start = '2018-01-01' if regular_before else '2020-01-01'
+            self._form_get_allocation_sample(leave_type_day, regular_alloc_start, number_of_days=10)
+            # Max number of leaves for the only level of the accrual plan is 10 + 10 for the regular allocation
+            self.assert_virtual_leaves_equal(leave_type_day, 20, self.employee_emp)
+            self.assert_virtual_leaves_equal(leave_type_day, 20, self.employee_emp, date='2022-02-01')
+
+            # 10 days leave
+            self._take_leave(leave_type_day, '2022-01-03', '2022-01-14')
+            # 10 days leave
+            self._take_leave(leave_type_day, '2023-01-02', '2023-01-13')
+            self.assert_virtual_leaves_equal(leave_type_day, 12, self.employee_emp, date='2023-02-01')
+            if add_future_leave:
+                # 10 days leaves that shouldn't be taken into account in this test
+                self._take_leave(leave_type_day, '2023-10-06', '2023-10-17')
+
+            # Right after spending all the 10 leaves ('2023-01-02' -> '2023-01-13')
+            # 10 days - 10 days (leave) + 2 days (1 month accrual)
+            self.assert_virtual_leaves_equal(leave_type_day, 12, self.employee_emp, date='2023-02-01')
```

Comment:
what's the point of checking the same value twice when `add_future_leave` is true? Looks like the argument serves no purpose an you could just add the leave any way, dividing the amount of tests by 2